### PR TITLE
Bring back clean-up and fix linking

### PR DESCRIPTION
**Pull request by lindig**
We had reverted a previous clean-up because of linking failure. This brings it back together with a fix. The fix is to add `xenctrl` as a dependency to the C stubs.

### CODE REVIEW

File: dbgring/dune [hunk 1]
```diff
@@ -5,6 +5,7 @@
  (libraries
    xapi-xenopsd
    xenctrl
+   xenctrl.xenmmap
```
Comment (psafont) [inline dbgring/dune:8]: This library is not present when building the rpm

**Comment by lindig** [comment]
I don't know what the problem with the sign-off is - both commits are singed off.

File: xc/dune [hunk 2]
```diff
@@ -1,43 +1,141 @@
...
+  xapi-rrd
+  xapi-stdext-date
+  xapi-stdext-pervasives
+  xapi-stdext-std
+  xapi-stdext-threads
+  xapi-stdext-unix
+  xapi-xenopsd
+  xapi-xenopsd.c_stubs
+  xenctrl
+  xenctrl.xenbus
```
Comment (psafont) [inline xc/dune:49]: This one isn't available either when building the rpm

**Comment by psafont** [comment]
The build scripts also depend on `scripts/make-custom-xenopsd.conf` being executable.

**Pull request merged**